how_tos/how_to_add_dynamical_decoupling.ipynb

```diff
@@ -0,0 +1,442 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "9750e811-ef28-4cd6-9d4c-95cdef31922e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "\n",
+    "from qiskit import transpile\n",
+    "from qiskit.circuit.library import QAOAAnsatz, CXGate\n",
+    "from qiskit.quantum_info import SparsePauliOp\n",
+    "from qiskit.transpiler.passes.routing.commuting_2q_gate_routing import SwapStrategy\n",
+    "\n",
+    "from qiskit_ibm_provider import IBMProvider\n",
+    "\n",
+    "from qopt_best_practices.swap_strategies import create_qaoa_swap_circuit\n",
+    "from qopt_best_practices.utils import load_edge_coloring"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "fd5c7e76-a9cb-430e-9dc9-aba93e50424b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open(\"data/graph_hardware_native_127nodes_001.json\", \"r\") as fin:\n",
+    "    data = json.load(fin)\n",
+    "\n",
+    "cost_operator = SparsePauliOp.from_list(data[\"paulis\"])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "ed94423b-2e51-4cda-97c8-de587b6119f5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "provider = IBMProvider(instance=\"quantum-demonstrations/main/locc\")\n",
+    "backend = provider.get_backend(\"ibm_kyiv\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "312e8cad-a9c8-4ad4-a3df-21073859ba3c",
+   "metadata": {},
+   "source": [
+    "## Circuit creation\n",
+    "\n",
+    "The problem is hardware native. Therefor, if we transpile its QAOA circuit to a 127 qubit backend with CNOT gates and a heavy-hexagonal coupling map then we expect to have (i) 288 CX gates per QAOA layer because a 127 qubit backend has 144 edges in its coupling map and each `RZZGate` requires two CNOT gates, and (ii) a CNOT depth of 6 because the heavy-hex coupling map has a maximum node degree of three. We will first construct a circuit with the standard Qiskit transpilation flow and show that it is sub-optimal."
```

Comment:
```suggestion
    "The problem is hardware native. Therefore, if we transpile its QAOA circuit to a 127 qubit backend with CNOT gates and a heavy-hexagonal coupling map then we expect to have (i) 288 CX gates per QAOA layer because a 127 qubit backend has 144 edges in its coupling map and each `RZZGate` requires two CNOT gates, and (ii) a CNOT depth of 6 because the heavy-hex coupling map has a maximum node degree of three. We will first construct a circuit with the standard Qiskit transpilation flow and show that it is sub-optimal."
```